cpp/include/utilities/cython.hpp

```diff
@@ -15,53 +15,113 @@
+    experimental::graph_view_t<int, int, float, true, false>* graph_view_t_float_transposed_ptr;
+    experimental::graph_view_t<int, int, double, true, false>* graph_view_t_double_transposed_ptr;
+    experimental::graph_view_t<int, int, float, true, true>* graph_view_t_float_mg_transposed_ptr;
+    experimental::graph_view_t<int, int, double, true, true>* graph_view_t_double_mg_transposed_ptr;
+  };
+
+  inline graph_container_t() : graph_ptr_union{nullptr}, graph_ptr_type{graphTypeEnum::null} {}
+
+  inline ~graph_container_t()
+  {
+    switch (graph_ptr_type) {
+      case graphTypeEnum::GraphCSRViewFloat: delete graph_ptr_union.GraphCSRViewFloatPtr; break;
+      case graphTypeEnum::GraphCSRViewDouble: delete graph_ptr_union.GraphCSRViewDoublePtr; break;
+      case graphTypeEnum::graph_view_t_float: delete graph_ptr_union.graph_view_t_float_ptr; break;
+      case graphTypeEnum::graph_view_t_double:
+        delete graph_ptr_union.graph_view_t_double_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_float_mg:
+        delete graph_ptr_union.graph_view_t_float_mg_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_double_mg:
+        delete graph_ptr_union.graph_view_t_double_mg_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_float_transposed:
+        delete graph_ptr_union.graph_view_t_float_transposed_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_double_transposed:
+        delete graph_ptr_union.graph_view_t_double_transposed_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_float_mg_transposed:
+        delete graph_ptr_union.graph_view_t_float_mg_transposed_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_double_mg_transposed:
+        delete graph_ptr_union.graph_view_t_double_mg_transposed_ptr;
+        break;
+      default: break;
+    }
+    graph_ptr_type = graphTypeEnum::null;
+  }
+
+  graphPtrUnion graph_ptr_union;
+  graphTypeEnum graph_ptr_type;
 };
 
 // Factory function for creating graph containers from basic types
-// FIXME: This should accept void* for offsets and indices as well and take a
-//        dtype directly for each instead of the enum/int.
-graph_container_t create_graph_t(raft::handle_t const& handle,
-                                 int* offsets,
-                                 int* indices,
-                                 void* weights,
-                                 int weightType,
-                                 int num_vertices,
-                                 int num_edges,
-                                 int* local_vertices,
-                                 int* local_edges,
-                                 int* local_offsets,
-                                 bool transposed,
-                                 bool multi_gpu);
+// FIXME: Should local_* values be void* as well?
+void create_graph_t(graph_container_t& graph_container,
```

Comment:
Add documentation for this function.

Would be useful to understand what local_vertices, local_edges and local_offsets mean while looking at the header.

Reply:
done